### ISSUE TITLE
shouldHideNavigationOnTap not working

### ISSUE THREAD

**Issue by mgiroux** [FolioReader/FolioReaderKit#268]
There was a previous bug that was flagged as fixed but is still present in 1.2.0

if you set shouldHideNavigationOnTap to true.. the navigation never shows up. if you set it to false the navigation is shown but will never go away on tap.

Thanks

**Comment by SolomiyaN** [comment]
Got this issue as well. This happens in iOS 11 because `toggleBars()` method in `FolioReaderCenter` is being called twice — first, because tap explicitly calls it, and the second time because showing a bar moves the content down (in iOS 11 only), so `scrollViewDidScroll` gets called. So it does show up, although it hides too quickly to even notice it.

I do not have a ready solution for it so far, but just maybe this helps.

**Comment by hebertialmeida** [comment]
@SolomiyaN Thanks for the details, I will take a look on that.

**Comment by zlyunduan** [comment]
In iOS 11, the `SafeAreaInset` causes Interface offset 64 When the `NavigationBar` is displayed, then the `scrollViewDidScroll` method is triggered, which in turn causes `NavigationBar` to hide again.
```
if #available(iOS 11.0, *) {
            self.additionalSafeAreaInsets = UIEdgeInsets.init(top: -64, left: 0, bottom: 0, right: 0)
        } else {
            // Fallback on earlier versions
        }
```

Add this to fix the problem. @hebertialmeida @SolomiyaN

**Comment by SolomiyaN** [comment]
Thanks @zlyunduan, I completely missed it! You saved the day!

**Comment by mgiroux** [comment]
in what file / line number?

**Comment by SolomiyaN** [comment]
@mgiroux I've put it in `viewDidLoad` method in `FolioReaderCenter` class.

**Comment by mgiroux** [comment]
it sorta works, thanks (by sorta i mean when you touch the screen it shows/hides but when you press on a button in the bar it should not hide but it does.. not a big deal)

**Comment by SolomiyaN** [comment]
@mgiroux it works exactly as it does on iOS 10.

**Comment by florent2009** [comment]
@zlyunduan Thank you you save my day as well !!!

**Comment by hebertialmeida** [comment]
@mgiroux @SolomiyaN @zlyunduan Is anyone able to properly fix this for iOS 11 and iPhone X but keeping the compatibility with iOS 10 and regular screen devices?

**Comment by hebertialmeida** [comment]
Thanks to @ryan-w this is now fixed in a better way on #301
